datalad/distribution/create_sibling.py

```diff
@@ -392,7 +393,14 @@ def _create_dataset_sibling(
 def _ls_remote_path(ssh, path):
     try:
         # yoh tried ls on mac
-        out, err = ssh("ls -A1 {}".format(sh_quote(path)))
+        # escape path explicitly with shlex.quote as 'sh' is a POSIX shell and
+        # sh_quote could decide to quote Windows-style
+        ls_cmd = "LC_ALL=C; export LC_ALL; /bin/ls -A1 {}".format(shlex.quote(path))
```

Comment:
```suggestion
        ls_cmd = "LC_ALL=C /bin/ls -A1 {}".format(shlex.quote(path))
```
isn't sufficient?

Reply:
No, some shellx don't support the `export VAR=value` directly AFAIK. I'd leave it like this.

Reply:
It followed a suggestion by @mih from the dev call, maybe he can chime in?

Reply:
@nobodyinperson already mentioned that the safe approach (as in most widely compatible with particular shells) is the one currently implemented. I have nothing else to add.